plugins/catalog-backend/src/service/NextRouter.ts

```diff
@@ -77,6 +105,13 @@ export async function createNextRouter(
 
   if (entitiesCatalog) {
     router
+      .use(
+        createPermissionIntegrationRouter({
```

Comment:
Now this integration ended up clean 👌

I wonder if should even inline `getEntity` as well? Having that defined as a top-level thing in this file risks getting a bit weird. Perhaps `getEntityResource` or something like that could do to

Reply:
I like `getEntityResource` - I'll also relocate this function to the bottom of the file to make its job as a helper more clear.